File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/viewmodel/AuthenticationViewModel.java
package com.skedgo.tripkit.booking.viewmodel;

import com.skedgo.tripkit.common.rx.Var;
import io.reactivex.Observable;

public interface AuthenticationViewModel {
  Var<String> url();
  Observable<Boolean> isSuccessful();
  Observable<Boolean> verify(Object param);
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/viewmodel/AuthenticationViewModel.kt
package com.skedgo.tripkit.booking.viewmodel

import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject

interface AuthenticationViewModel {
    fun url(): BehaviorSubject<String>

    fun isSuccessful(): Observable<Boolean>

    fun verify(param: Any): Observable<Boolean>
}